From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Garrett Gu <garrett@cloudflare.com>
Date: Mon, 22 Apr 2024 10:10:21 -0500
Subject: Modify where to look for fp16 dependency. This dependency is normally
 downloaded by gn but we need to fetch it in Bazel. We do so in the workerd
 repo in the WORKSPACE file.

Signed-off-by: James M Snell <jsnell@cloudflare.com>

diff --git a/BUILD.bazel b/BUILD.bazel
index 29df31fa21b2addd10a1d6b9ed450ee585dd42f7..4abddcdb7579d15600231a5093c08252ca358366 100644
--- a/BUILD.bazel
+++ b/BUILD.bazel
@@ -4205,17 +4205,23 @@ v8_library(
     ],
 )
 
-v8_library(
+cc_library(
     name = "lib_fp16",
-    srcs = ["third_party/fp16/src/include/fp16.h"],
     hdrs = [
-        "third_party/fp16/src/include/fp16/fp16.h",
-        "third_party/fp16/src/include/fp16/bitcasts.h",
-        "third_party/fp16/src/include/fp16/macros.h",
+        "@fp16//:include/fp16.h",
     ],
-    includes = [
-        "third_party/fp16/src/include",
+    include_prefix = "third_party/fp16/src",
+    deps = [":lib_fp16_includes"],
+)
+
+cc_library(
+    name = "lib_fp16_includes",
+    hdrs = [
+        "@fp16//:include/fp16/bitcasts.h",
+        "@fp16//:include/fp16/macros.h",
+        "@fp16//:include/fp16/fp16.h",
     ],
+    strip_include_prefix = "include",
 )
 
 filegroup(
@@ -4515,7 +4521,6 @@ filegroup(
         "third_party/utf8-decoder/**/*.h",
         "third_party/simdutf/**/*.h",
         "third_party/rapidhash-v8/**/*.h",
-        "third_party/fp16/**/*.h",
         "third_party/wasm-api/**/*.h",
         "third_party/siphash/**/*.h",
         "third_party/vtune/**/*.h",
